src/components/Message.vue

```diff
@@ -44,7 +44,7 @@
 						</ActionButton>
 
 						<ActionButton icon="icon-junk" @click="onToggleJunk">
-							{{ envelope.flags.notjunk ? t('mail', 'Mark not junk') : t('mail', 'Mark junk') }}
+							{{ envelope.flags.notjunk ? t('mail', 'Mark not spam') : t('mail', 'Mark as spam') }}
```

Comment:
Shouldn't this also be `.flags.junk` like in 
https://github.com/nextcloud/mail/blob/fcf2fd0b444af2f8fb9b03fff18d324f5d10ed8e/src/components/Envelope.vue#L59

I could find occurrences of `notjunk` in the server, but this is the only place where its used in js/vue.

Reply:
> Shouldn't this also be `.flags.junk` like in

Yes it should. Though technically both keywords might be used on IMAP …

Reply:
Fixed

Reply:
I think would also explain why @blizzz experienced that the Junk state does not change as you click the button. Because the `notjunk` is not there, so it's not a reactive object in Vue terms

Reply:
Yep, this fixes the reactivity bug :)